tests/test_welltest_dpds.py

```diff
@@ -182,3 +206,50 @@ def test_get_weighted_avg_press_time_derivative_lag2():
     assert len(dpdspt_w_lag2) == 247
     assert dpdspt_w_lag2[0] == pytest.approx(0.43083638)
     assert dpdspt_w_lag2[-1] == pytest.approx(0.12729989)
+
+
+def test_genobs_vec(tmpdir):
+    """ Test genobs_vec """
+
+    tmpdir.chdir()
+    mockcsv = """
+    Time\tdTime
+    (hr)\t(hr)
+    0\t0
+    1\t1
+    """
+    Path("index.txt").write_text(mockcsv)
+
+    vec = np.array([0, 0.5, 1, 2])
+    time = np.array([0, 1, 2, 3])
+    genobs_vec = welltest_dpds.genobs_vec("index.txt", vec, time)
+
+    assert len(genobs_vec) == 2
+    assert genobs_vec[1] == pytest.approx(0.5)
+
+
+def test_to_csv(tmpdir):
+    """ Test to_csv """
+
+    tmpdir.chdir()
+
+    vec = np.array([0, 0.5, 1, 2])
+    welltest_dpds.to_csv("mock.csv", [vec])
+    assert os.path.exists("mock.csv")
```

Comment:
subscript went `pathlib`-only one day ago ;)

Reply:
Point taken! I have now replaces all traces of os with pathlib in the test module :)